lib/workers/repository/process/fetch.ts

```diff
@@ -60,102 +64,145 @@ async function fetchDepUpdates(
+      if (
+        packageFileConfig.repoIsOnboarded === true ||
+        !(err instanceof ExternalHostError)
+      ) {
+        return Result.err(err);
+      }
+
+      const cause = err.err;
+      return Result.ok({
+        updates: [],
+        warnings: [
+          { topic: 'Lookup Error', message: `${depName}: ${cause.message}` },
+        ],
+      });
+    })
+    .transform((upd): PackageDependency => Object.assign(dep, upd));
+
+  const duration = Date.now() - start;
+  const lookups = memCache.get<LookupStats[]>('lookup-stats') || [];
+  lookups.push({ datasource: depConfig.datasource, duration });
+  memCache.set('lookup-stats', lookups);
+
+  return result;
 }
 
-async function fetchManagerUpdates(
+function createLookupTasks(
   config: RenovateConfig,
-  packageFiles: Record<string, PackageFile[]>,
-  manager: string,
-): Promise<void> {
-  const managerConfig = getManagerConfig(config, manager);
-  const queue = packageFiles[manager].map(
-    (pFile) => (): Promise<void> =>
-      fetchManagerPackagerFileUpdates(config, managerConfig, pFile),
-  );
-  logger.trace(
-    { manager, queueLength: queue.length },
-    'fetchManagerUpdates starting',
-  );
-  await p.all(queue);
-  logger.trace({ manager }, 'fetchManagerUpdates finished');
+  managerPackageFiles: Record<string, PackageFile[]>,
+): LookupTask[] {
+  const lookupTasks: LookupTask[] = [];
+
+  for (const [manager, packageFiles] of Object.entries(managerPackageFiles)) {
+    const managerConfig = getManagerConfig(config, manager);
+
+    for (const packageFile of packageFiles) {
+      const packageFileConfig = mergeChildConfig(managerConfig, packageFile);
+      if (packageFile.extractedConstraints) {
+        packageFileConfig.constraints = {
+          ...packageFile.extractedConstraints,
+          ...config.constraints,
+        };
+      }
+
+      for (const dep of packageFile.deps) {
+        const lookupTask: LookupTask = async () => {
+          const result = await lookup(packageFileConfig, dep);
+          return {
+            packageFileName: packageFile.packageFile,
+            manager: managerConfig.manager,
+            result,
+          };
+        };
+        lookupTasks.push(lookupTask);
+      }
+    }
+  }
+
+  return lookupTasks;
 }
 
 export async function fetchUpdates(
   config: RenovateConfig,
-  packageFiles: Record<string, PackageFile[]>,
+  managerPackageFiles: Record<string, PackageFile[]>,
 ): Promise<void> {
-  const managers = Object.keys(packageFiles);
-  const allManagerJobs = managers.map((manager) =>
-    fetchManagerUpdates(config, packageFiles, manager),
+  logger.debug(
+    { baseBranch: config.baseBranch },
+    'Starting package releases lookups',
   );
-  await Promise.all(allManagerJobs);
-  PackageFiles.add(config.baseBranch!, { ...packageFiles });
+
+  const allTasks = createLookupTasks(config, managerPackageFiles);
+
+  const fetchResults = await p.all(allTasks, { concurrency: Infinity });
```

Comment:
use promise.all or allsettled, we don't need pMap here if we do infinite parallel calls.